pkg/lifecycle/service.go

```diff
@@ -154,33 +216,43 @@ func (s *Service) stopGraceful(ctx context.Context, pl *Instance, reason error)
 			}
 		}
 	}
+
+	if len(errs) == 0 {
+		s.runningPipelines[rp.pipeline.ID] = nil
+		return nil
+	}
+
 	return cerrors.Join(errs...)
 }
 
-func (s *Service) stopForceful(ctx context.Context, pl *Instance) error {
+func (s *Service) stopForceful(ctx context.Context, rp *runnablePipeline) error {
 	s.logger.Info(ctx).
-		Str(log.PipelineIDField, pl.ID).
-		Any(log.PipelineStatusField, pl.GetStatus()).
+		Str(log.PipelineIDField, rp.pipeline.ID).
+		Any(log.PipelineStatusField, rp.pipeline.GetStatus()).
 		Msg("force stopping pipeline")
-	pl.t.Kill(ErrForceStop)
-	for _, n := range pl.n {
+	rp.t.Kill(pipeline.ErrForceStop)
+	for _, n := range rp.n {
 		if node, ok := n.(stream.ForceStoppableNode); ok {
 			// stop all pub nodes
 			s.logger.Trace(ctx).Str(log.NodeIDField, n.ID()).Msg("force stopping node")
 			node.ForceStop(ctx)
 		}
 	}
+
+	s.runningPipelines[rp.pipeline.ID] = nil
```

Comment:
We should rather delete the key entirely.

```suggestion
	delete(s.runningPipelines, rp.pipeline.ID)
```